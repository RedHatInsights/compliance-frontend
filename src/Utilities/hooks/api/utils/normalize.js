import _ from 'lodash';
import normalize from 'json-api-normalizer';

export const findItemById = (id, array) =>
  array.find((item) => parseInt(item.id) === parseInt(id));

export const findIncluded = (payload, id, type) => {
  let singleType = type.slice(0, -1);
  let singleTypePayload = payload[singleType];
  let foundItem = singleTypePayload
    ? findItemById(id, Object.values(singleTypePayload))
    : {};
  return foundItem ? foundItem : {};
};

export const includeItemRelationships = (item, fullPayload) => {
  if (!item || !item.relationships) {
    return item;
  }

  const relatedResourceKinds = Object.keys(item.relationships);
  let relatedResources = {};

  relatedResourceKinds.forEach((relatedResourceKind) => {
    const relatedResourcesData = Object.values(
      item.relationships[relatedResourceKind].data
    );
    const relatedIds = relatedResourcesData.map((item) => item.id);
    if (!relatedResources[relatedResourceKind]) {
      relatedResources[relatedResourceKind] = {};
    }

    relatedIds.forEach((id) => {
      let itemData = includeItemRelationships(
        findItemById(id, relatedResourcesData),
        fullPayload
      );
      let additionalData = includeItemRelationships(
        findIncluded(fullPayload, id, relatedResourceKind),
        fullPayload
      );

      relatedResources[relatedResourceKind][id] = {
        ...itemData,
        ...additionalData,
      };
    });
  });
  return { ...item, ...relatedResources };
};

export const includeRelationships = (normalizedPayload) => {
  const resourcesKinds = Object.keys(normalizedPayload);
  let relationshipsIncluded = {};

  resourcesKinds.forEach((resourceKind) => {
    const resources = normalizedPayload[resourceKind];
    const resourceIds = Object.keys(resources);

    resourceIds.forEach((id) => {
      if (!relationshipsIncluded[resourceKind]) {
        relationshipsIncluded[resourceKind] = {};
      }

      relationshipsIncluded[resourceKind][id] = includeItemRelationships(
        resources[id],
        normalizedPayload
      );
    });
  });
  return relationshipsIncluded;
};

export const normalizePayload = (payload, endpoint) =>
  includeRelationships(normalize(payload, { endpoint }));

const compareValues = (key, order = 'asc') => {
  return function (a, b) {
        if (!a.hasOwnProperty(key) || !b.hasOwnProperty(key)) { // eslint-disable-line
      return 0;
    }

    const first = a[key];
    const second = b[key];

    let comparison = 0;
    if (first > second) {
      comparison = 1;
    } else if (first < second) {
      comparison = -1;
    }

    return order === 'desc' ? comparison * -1 : comparison;
  };
};

export const sortData = (data, sortBy) => {
  if (data && sortBy) {
    const sort = sortBy.split(' ');
    const sortedValues = Object.values(data).sort(
      compareValues(sort[0], sort[1])
    );
    data = Object.assign({}, sortedValues);
  }

  return data ? data : {};
};

export const normalizeData = (data, property, endpoint, sortBy) => {
  const normalizedData = _.mapValues(
    normalizePayload(data, endpoint)[property],
    (item) => ({ ...item, ...item.attributes })
  );
  return sortData(normalizedData, sortBy);
};
