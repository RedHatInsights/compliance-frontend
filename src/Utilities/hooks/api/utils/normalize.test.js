import supportedSsgs from '../__fixtures__/supported_ssgs.json';
import { normalizeData } from './normalize';

describe('normalizeData', () => {
  it('normalizes API data', () => {
    expect(
      normalizeData(supportedSsgs, 'supportedSsg', '/supported_ssgs')
    ).toMatchSnapshot();
  });

  //     it('sorts it when sortBy is given', () => {
  //         expect(normalizeData(supportedSsgs, 'supportedSsg', '/supported_ssgs', 'osMajorVersion desc')).toMatchSnapshot();
  //     });
  //
  //     it('sorts it asc when sortBy is given', () => {
  //         expect(normalizeData(supportedSsgs, 'supportedSsg', '/supported_ssgs', 'osMajorVersion asc')).toMatchSnapshot();
  //     });
  //
  //     it('skips sorting when column is not known', () => {
  //         expect(normalizeData(supportedSsgs, 'supportedSsg', '/supported_ssgs', 'unknownColumn asc')).toMatchSnapshot();
  //     });
});
