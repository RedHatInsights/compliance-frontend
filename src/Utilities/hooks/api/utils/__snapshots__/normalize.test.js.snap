// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeData normalizes API data 1`] = `
{
  "RHEL-6.10:scap-security-guide-0.1.28-4.el6": {
    "attributes": {
      "osMajorVersion": "6",
      "osMinorVersion": "10",
      "package": "scap-security-guide-0.1.28-4.el6",
      "profiles": [
        "C2S",
        "common",
        "CS2",
        "CSCF-RHEL6-MLS",
        "nist-cl-il-al",
        "pci-dss",
        "rht-ccp",
        "server",
        "standard",
        "stig-rhel6-server-upstream",
        "usgcb-rhel6-server",
      ],
      "version": "0.1.28",
    },
    "id": "RHEL-6.10:scap-security-guide-0.1.28-4.el6",
    "osMajorVersion": "6",
    "osMinorVersion": "10",
    "package": "scap-security-guide-0.1.28-4.el6",
    "profiles": [
      "C2S",
      "common",
      "CS2",
      "CSCF-RHEL6-MLS",
      "nist-cl-il-al",
      "pci-dss",
      "rht-ccp",
      "server",
      "standard",
      "stig-rhel6-server-upstream",
      "usgcb-rhel6-server",
    ],
    "type": "supportedSsg",
    "version": "0.1.28",
  },
  "RHEL-6.6:scap-security-guide-0.1.18-3.el6": {
    "attributes": {
      "osMajorVersion": "6",
      "osMinorVersion": "6",
      "package": "scap-security-guide-0.1.18-3.el6",
      "profiles": [
        "C2S",
        "common",
        "CS2",
        "CSCF-RHEL6-MLS",
        "rht-ccp",
        "server",
        "stig-rhel6-server-upstream",
        "test",
        "usgcb-rhel6-server",
      ],
      "version": "0.1.18",
    },
    "id": "RHEL-6.6:scap-security-guide-0.1.18-3.el6",
    "osMajorVersion": "6",
    "osMinorVersion": "6",
    "package": "scap-security-guide-0.1.18-3.el6",
    "profiles": [
      "C2S",
      "common",
      "CS2",
      "CSCF-RHEL6-MLS",
      "rht-ccp",
      "server",
      "stig-rhel6-server-upstream",
      "test",
      "usgcb-rhel6-server",
    ],
    "type": "supportedSsg",
    "version": "0.1.18",
  },
  "RHEL-6.7:scap-security-guide-0.1.21-3.el6": {
    "attributes": {
      "osMajorVersion": "6",
      "osMinorVersion": "7",
      "package": "scap-security-guide-0.1.21-3.el6",
      "profiles": [
        "C2S",
        "common",
        "CS2",
        "CSCF-RHEL6-MLS",
        "rht-ccp",
        "server",
        "stig-rhel6-server-upstream",
        "usgcb-rhel6-server",
      ],
      "version": "0.1.21",
    },
    "id": "RHEL-6.7:scap-security-guide-0.1.21-3.el6",
    "osMajorVersion": "6",
    "osMinorVersion": "7",
    "package": "scap-security-guide-0.1.21-3.el6",
    "profiles": [
      "C2S",
      "common",
      "CS2",
      "CSCF-RHEL6-MLS",
      "rht-ccp",
      "server",
      "stig-rhel6-server-upstream",
      "usgcb-rhel6-server",
    ],
    "type": "supportedSsg",
    "version": "0.1.21",
  },
  "RHEL-6.8:scap-security-guide-0.1.28-2.el6": {
    "attributes": {
      "osMajorVersion": "6",
      "osMinorVersion": "8",
      "package": "scap-security-guide-0.1.28-2.el6",
      "profiles": [
        "C2S",
        "common",
        "CS2",
        "CSCF-RHEL6-MLS",
        "nist-cl-il-al",
        "pci-dss",
        "rht-ccp",
        "server",
        "standard",
        "stig-rhel6-server-upstream",
        "usgcb-rhel6-server",
      ],
      "version": "0.1.28",
    },
    "id": "RHEL-6.8:scap-security-guide-0.1.28-2.el6",
    "osMajorVersion": "6",
    "osMinorVersion": "8",
    "package": "scap-security-guide-0.1.28-2.el6",
    "profiles": [
      "C2S",
      "common",
      "CS2",
      "CSCF-RHEL6-MLS",
      "nist-cl-il-al",
      "pci-dss",
      "rht-ccp",
      "server",
      "standard",
      "stig-rhel6-server-upstream",
      "usgcb-rhel6-server",
    ],
    "type": "supportedSsg",
    "version": "0.1.28",
  },
  "RHEL-6.9:scap-security-guide-0.1.28-3.el6": {
    "attributes": {
      "osMajorVersion": "6",
      "osMinorVersion": "9",
      "package": "scap-security-guide-0.1.28-3.el6",
      "profiles": [
        "C2S",
        "common",
        "CS2",
        "CSCF-RHEL6-MLS",
        "nist-cl-il-al",
        "pci-dss",
        "rht-ccp",
        "server",
        "standard",
        "stig-rhel6-server-upstream",
        "usgcb-rhel6-server",
      ],
      "version": "0.1.28",
    },
    "id": "RHEL-6.9:scap-security-guide-0.1.28-3.el6",
    "osMajorVersion": "6",
    "osMinorVersion": "9",
    "package": "scap-security-guide-0.1.28-3.el6",
    "profiles": [
      "C2S",
      "common",
      "CS2",
      "CSCF-RHEL6-MLS",
      "nist-cl-il-al",
      "pci-dss",
      "rht-ccp",
      "server",
      "standard",
      "stig-rhel6-server-upstream",
      "usgcb-rhel6-server",
    ],
    "type": "supportedSsg",
    "version": "0.1.28",
  },
  "RHEL-7.1:scap-security-guide-0.1.19-2.el7": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "1",
      "package": "scap-security-guide-0.1.19-2.el7",
      "profiles": [
        "rht-ccp",
      ],
      "version": "0.1.19",
    },
    "id": "RHEL-7.1:scap-security-guide-0.1.19-2.el7",
    "osMajorVersion": "7",
    "osMinorVersion": "1",
    "package": "scap-security-guide-0.1.19-2.el7",
    "profiles": [
      "rht-ccp",
    ],
    "type": "supportedSsg",
    "version": "0.1.19",
  },
  "RHEL-7.2:scap-security-guide-0.1.25-3.el7": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "2",
      "package": "scap-security-guide-0.1.25-3.el7",
      "profiles": [
        "common",
        "pci-dss",
        "rht-ccp",
        "standard",
        "stig-rhel7-server-upstream",
      ],
      "version": "0.1.25",
    },
    "id": "RHEL-7.2:scap-security-guide-0.1.25-3.el7",
    "osMajorVersion": "7",
    "osMinorVersion": "2",
    "package": "scap-security-guide-0.1.25-3.el7",
    "profiles": [
      "common",
      "pci-dss",
      "rht-ccp",
      "standard",
      "stig-rhel7-server-upstream",
    ],
    "type": "supportedSsg",
    "version": "0.1.25",
  },
  "RHEL-7.3:scap-security-guide-0.1.30-5.el7_3": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "3",
      "package": "scap-security-guide-0.1.30-5.el7_3",
      "profiles": [
        "C2S",
        "cjis-rhel7-server",
        "common",
        "nist-cl-il-al",
        "ospp-rhel7-server",
        "pci-dss",
        "rht-ccp",
        "standard",
        "stig-rhel7-server-gui-upstream",
        "stig-rhel7-server-upstream",
        "stig-rhel7-workstation-upstream",
      ],
      "version": "0.1.30",
    },
    "id": "RHEL-7.3:scap-security-guide-0.1.30-5.el7_3",
    "osMajorVersion": "7",
    "osMinorVersion": "3",
    "package": "scap-security-guide-0.1.30-5.el7_3",
    "profiles": [
      "C2S",
      "cjis-rhel7-server",
      "common",
      "nist-cl-il-al",
      "ospp-rhel7-server",
      "pci-dss",
      "rht-ccp",
      "standard",
      "stig-rhel7-server-gui-upstream",
      "stig-rhel7-server-upstream",
      "stig-rhel7-workstation-upstream",
    ],
    "type": "supportedSsg",
    "version": "0.1.30",
  },
  "RHEL-7.4:scap-security-guide-0.1.33-6.el7_4": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "4",
      "package": "scap-security-guide-0.1.33-6.el7_4",
      "profiles": [
        "C2S",
        "cjis-rhel7-server",
        "common",
        "docker-host",
        "nist-800-171-cui",
        "ospp-rhel7",
        "pci-dss_centric",
        "pci-dss",
        "rht-ccp",
        "standard",
        "stig-rhel7-disa",
        "stig-rhevh-upstream",
      ],
      "version": "0.1.33",
    },
    "id": "RHEL-7.4:scap-security-guide-0.1.33-6.el7_4",
    "osMajorVersion": "7",
    "osMinorVersion": "4",
    "package": "scap-security-guide-0.1.33-6.el7_4",
    "profiles": [
      "C2S",
      "cjis-rhel7-server",
      "common",
      "docker-host",
      "nist-800-171-cui",
      "ospp-rhel7",
      "pci-dss_centric",
      "pci-dss",
      "rht-ccp",
      "standard",
      "stig-rhel7-disa",
      "stig-rhevh-upstream",
    ],
    "type": "supportedSsg",
    "version": "0.1.33",
  },
  "RHEL-7.5:scap-security-guide-0.1.36-10.el7_5": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "5",
      "package": "scap-security-guide-0.1.36-10.el7_5",
      "profiles": [
        "C2S",
        "cjis-rhel7-server",
        "common",
        "docker-host",
        "nist-800-171-cui",
        "ospp-rhel7",
        "pci-dss_centric",
        "pci-dss",
        "rht-ccp",
        "standard",
        "stig-rhel7-disa",
        "stig-rhevh-upstream",
      ],
      "version": "0.1.36",
    },
    "id": "RHEL-7.5:scap-security-guide-0.1.36-10.el7_5",
    "osMajorVersion": "7",
    "osMinorVersion": "5",
    "package": "scap-security-guide-0.1.36-10.el7_5",
    "profiles": [
      "C2S",
      "cjis-rhel7-server",
      "common",
      "docker-host",
      "nist-800-171-cui",
      "ospp-rhel7",
      "pci-dss_centric",
      "pci-dss",
      "rht-ccp",
      "standard",
      "stig-rhel7-disa",
      "stig-rhevh-upstream",
    ],
    "type": "supportedSsg",
    "version": "0.1.36",
  },
  "RHEL-7.6:scap-security-guide-0.1.40-13.el7_6": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "6",
      "package": "scap-security-guide-0.1.40-13.el7_6",
      "profiles": [
        "C2S",
        "cjis",
        "hipaa",
        "nist-800-171-cui",
        "ospp42",
        "ospp",
        "pci-dss_centric",
        "pci-dss",
        "rht-ccp",
        "standard",
        "stig-rhel7-disa",
      ],
      "version": "0.1.40",
    },
    "id": "RHEL-7.6:scap-security-guide-0.1.40-13.el7_6",
    "osMajorVersion": "7",
    "osMinorVersion": "6",
    "package": "scap-security-guide-0.1.40-13.el7_6",
    "profiles": [
      "C2S",
      "cjis",
      "hipaa",
      "nist-800-171-cui",
      "ospp42",
      "ospp",
      "pci-dss_centric",
      "pci-dss",
      "rht-ccp",
      "standard",
      "stig-rhel7-disa",
    ],
    "type": "supportedSsg",
    "version": "0.1.40",
  },
  "RHEL-7.7:scap-security-guide-0.1.43-13.el7": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "7",
      "package": "scap-security-guide-0.1.43-13.el7",
      "profiles": [
        "C2S",
        "cjis",
        "hipaa",
        "nist-800-171-cui",
        "ospp42",
        "ospp",
        "pci-dss_centric",
        "pci-dss",
        "rhelh-vpp",
        "rht-ccp",
        "standard",
        "stig-rhel7-disa",
      ],
      "version": "0.1.43",
    },
    "id": "RHEL-7.7:scap-security-guide-0.1.43-13.el7",
    "osMajorVersion": "7",
    "osMinorVersion": "7",
    "package": "scap-security-guide-0.1.43-13.el7",
    "profiles": [
      "C2S",
      "cjis",
      "hipaa",
      "nist-800-171-cui",
      "ospp42",
      "ospp",
      "pci-dss_centric",
      "pci-dss",
      "rhelh-vpp",
      "rht-ccp",
      "standard",
      "stig-rhel7-disa",
    ],
    "type": "supportedSsg",
    "version": "0.1.43",
  },
  "RHEL-7.8:scap-security-guide-0.1.46-11.el7": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "8",
      "package": "scap-security-guide-0.1.46-11.el7",
      "profiles": [
        "anssi_nt28_enhanced",
        "anssi_nt28_high",
        "anssi_nt28_intermediary",
        "anssi_nt28_minimal",
        "C2S",
        "cjis",
        "cui",
        "e8",
        "hipaa",
        "ncp",
        "ospp",
        "pci-dss_centric",
        "pci-dss",
        "rhelh-stig",
        "rhelh-vpp",
        "rht-ccp",
        "standard",
        "stig",
      ],
      "version": "0.1.46",
    },
    "id": "RHEL-7.8:scap-security-guide-0.1.46-11.el7",
    "osMajorVersion": "7",
    "osMinorVersion": "8",
    "package": "scap-security-guide-0.1.46-11.el7",
    "profiles": [
      "anssi_nt28_enhanced",
      "anssi_nt28_high",
      "anssi_nt28_intermediary",
      "anssi_nt28_minimal",
      "C2S",
      "cjis",
      "cui",
      "e8",
      "hipaa",
      "ncp",
      "ospp",
      "pci-dss_centric",
      "pci-dss",
      "rhelh-stig",
      "rhelh-vpp",
      "rht-ccp",
      "standard",
      "stig",
    ],
    "type": "supportedSsg",
    "version": "0.1.46",
  },
  "RHEL-7.9:scap-security-guide-0.1.49-13.el7": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "9",
      "package": "scap-security-guide-0.1.49-13.el7",
      "profiles": [
        "anssi_nt28_enhanced",
        "anssi_nt28_high",
        "anssi_nt28_intermediary",
        "anssi_nt28_minimal",
        "C2S",
        "cis",
        "cjis",
        "cui",
        "e8",
        "hipaa",
        "ncp",
        "ospp",
        "pci-dss_centric",
        "pci-dss",
        "rhelh-stig",
        "rhelh-vpp",
        "rht-ccp",
        "standard",
        "stig",
      ],
      "version": "0.1.49",
    },
    "id": "RHEL-7.9:scap-security-guide-0.1.49-13.el7",
    "osMajorVersion": "7",
    "osMinorVersion": "9",
    "package": "scap-security-guide-0.1.49-13.el7",
    "profiles": [
      "anssi_nt28_enhanced",
      "anssi_nt28_high",
      "anssi_nt28_intermediary",
      "anssi_nt28_minimal",
      "C2S",
      "cis",
      "cjis",
      "cui",
      "e8",
      "hipaa",
      "ncp",
      "ospp",
      "pci-dss_centric",
      "pci-dss",
      "rhelh-stig",
      "rhelh-vpp",
      "rht-ccp",
      "standard",
      "stig",
    ],
    "type": "supportedSsg",
    "version": "0.1.49",
  },
  "RHEL-7.9:scap-security-guide-0.1.52-2.el7_9": {
    "attributes": {
      "osMajorVersion": "7",
      "osMinorVersion": "9",
      "package": "scap-security-guide-0.1.52-2.el7_9",
      "profiles": [
        "anssi_nt28_enhanced",
        "anssi_nt28_high",
        "anssi_nt28_intermediary",
        "anssi_nt28_minimal",
        "C2S",
        "cis",
        "cjis",
        "cui",
        "e8",
        "hipaa",
        "ncp",
        "ospp",
        "pci-dss_centric",
        "pci-dss",
        "rhelh-stig",
        "rhelh-vpp",
        "rht-ccp",
        "standard",
        "stig",
      ],
      "version": "0.1.52",
    },
    "id": "RHEL-7.9:scap-security-guide-0.1.52-2.el7_9",
    "osMajorVersion": "7",
    "osMinorVersion": "9",
    "package": "scap-security-guide-0.1.52-2.el7_9",
    "profiles": [
      "anssi_nt28_enhanced",
      "anssi_nt28_high",
      "anssi_nt28_intermediary",
      "anssi_nt28_minimal",
      "C2S",
      "cis",
      "cjis",
      "cui",
      "e8",
      "hipaa",
      "ncp",
      "ospp",
      "pci-dss_centric",
      "pci-dss",
      "rhelh-stig",
      "rhelh-vpp",
      "rht-ccp",
      "standard",
      "stig",
    ],
    "type": "supportedSsg",
    "version": "0.1.52",
  },
  "RHEL-8.0:scap-security-guide-0.1.42-11.el8": {
    "attributes": {
      "osMajorVersion": "8",
      "osMinorVersion": "0",
      "package": "scap-security-guide-0.1.42-11.el8",
      "profiles": [
        "ospp",
        "pci-dss",
      ],
      "version": "0.1.42",
    },
    "id": "RHEL-8.0:scap-security-guide-0.1.42-11.el8",
    "osMajorVersion": "8",
    "osMinorVersion": "0",
    "package": "scap-security-guide-0.1.42-11.el8",
    "profiles": [
      "ospp",
      "pci-dss",
    ],
    "type": "supportedSsg",
    "version": "0.1.42",
  },
  "RHEL-8.1:scap-security-guide-0.1.46-1.el8": {
    "attributes": {
      "osMajorVersion": "8",
      "osMinorVersion": "1",
      "package": "scap-security-guide-0.1.46-1.el8",
      "profiles": [
        "ospp",
        "pci-dss",
      ],
      "version": "0.1.46",
    },
    "id": "RHEL-8.1:scap-security-guide-0.1.46-1.el8",
    "osMajorVersion": "8",
    "osMinorVersion": "1",
    "package": "scap-security-guide-0.1.46-1.el8",
    "profiles": [
      "ospp",
      "pci-dss",
    ],
    "type": "supportedSsg",
    "version": "0.1.46",
  },
  "RHEL-8.1:scap-security-guide-0.1.47-8.el8_1": {
    "attributes": {
      "osMajorVersion": "8",
      "osMinorVersion": "1",
      "package": "scap-security-guide-0.1.47-8.el8_1",
      "profiles": [
        "ospp",
        "pci-dss",
      ],
      "version": "0.1.47",
    },
    "id": "RHEL-8.1:scap-security-guide-0.1.47-8.el8_1",
    "osMajorVersion": "8",
    "osMinorVersion": "1",
    "package": "scap-security-guide-0.1.47-8.el8_1",
    "profiles": [
      "ospp",
      "pci-dss",
    ],
    "type": "supportedSsg",
    "version": "0.1.47",
  },
  "RHEL-8.2:scap-security-guide-0.1.48-10.el8_2": {
    "attributes": {
      "osMajorVersion": "8",
      "osMinorVersion": "2",
      "package": "scap-security-guide-0.1.48-10.el8_2",
      "profiles": [
        "e8",
        "ospp",
        "pci-dss",
        "stig",
      ],
      "version": "0.1.48",
    },
    "id": "RHEL-8.2:scap-security-guide-0.1.48-10.el8_2",
    "osMajorVersion": "8",
    "osMinorVersion": "2",
    "package": "scap-security-guide-0.1.48-10.el8_2",
    "profiles": [
      "e8",
      "ospp",
      "pci-dss",
      "stig",
    ],
    "type": "supportedSsg",
    "version": "0.1.48",
  },
  "RHEL-8.3:scap-security-guide-0.1.50-16.el8_3": {
    "attributes": {
      "osMajorVersion": "8",
      "osMinorVersion": "3",
      "package": "scap-security-guide-0.1.50-16.el8_3",
      "profiles": [
        "cis",
        "cui",
        "e8",
        "hipaa",
        "ospp",
        "pci-dss",
        "stig",
      ],
      "version": "0.1.50",
    },
    "id": "RHEL-8.3:scap-security-guide-0.1.50-16.el8_3",
    "osMajorVersion": "8",
    "osMinorVersion": "3",
    "package": "scap-security-guide-0.1.50-16.el8_3",
    "profiles": [
      "cis",
      "cui",
      "e8",
      "hipaa",
      "ospp",
      "pci-dss",
      "stig",
    ],
    "type": "supportedSsg",
    "version": "0.1.50",
  },
}
`;
